src/project_types.rs

```diff
@@ -1,17 +1,37 @@
 /// Structure to represent a Proton Project.
 /// This is what will be written to a Protonfile at the project root.
+
+#[derive(Debug, RustcEncodable, RustcDecodable)]
+pub struct User {
+    pub name: String,
+    pub public_key: String,
+}
+
 #[derive(Debug, RustcEncodable, RustcDecodable)]
 pub struct Project {
     pub name: String,
+    pub users: Vec<User>,
 }
 
 impl Project {
     /// Creates an empty project
     pub fn empty() -> Project {
         Project {
             name: "New Project".to_owned(),
+            users: Vec::new(),
         }
     }
+
+    // Adds a user to the project
+    pub fn add_user(&mut self, name: String, pub_key: String) {
```

Comment:
Immutable > mutable in the general case. Make this `fn add_user(&self, name: String, pub_key: String) -> Project`.